src/state/slices/validatorDataSlice/selectors.ts

```diff
@@ -0,0 +1,21 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { ReduxState } from 'state/reducer'
+
+import { PubKey } from './validatorDataSlice'
+
+export const selectValidatorAddress = (_state: ReduxState, validatorAddress: PubKey) =>
+  validatorAddress
+
+export const selectValidatorData = (state: ReduxState) => state.validatorData
+export const selectAllValidatorsData = createSelector(
+  selectValidatorData,
+  validatorData => validatorData.byValidator,
+)
+
+export const selectSingleValidator = createSelector(
+  selectValidatorData,
+  selectValidatorAddress,
+  (stakingData, validatorAddress) => {
+    return stakingData.byValidator[validatorAddress] || null
+  },
+)
```

Comment:
as a best practice can we please type the return of selectors? i know we have a lot that are strictly wrong already but want to be sure for new selectors that consumers can be confident in what they're receiving

Reply:
Agreed. These were having typings working implicitly but doesn't hurt to explicitly type them to ensure that we don't break things in the future by changing them to return a different type than the expected one.